### DISCUSSION TITLE
How to use global interrupts

### DISCUSSION THREAD

**Discussion by fcuzzocrea** [polarfire-soc/polarfire-soc-documentation#53] [How to] [answered]
Hi! 

I am trying to write a simple demo application to interact with UART interrupts using the baremetal library.

Right now, to keep it as simple at possible I am only interacting with HART 0.

If I use local interrupts on hart0 (i.e. using MSS_UART_enable_local_irq() on UART0) this simple program (https://hastebin.com/ucemunudik.cpp) works as expected, in the sense that if I type something on the keyboard from my UART output, the "hart0 MMUART0 local IRQ." is printed correctly.

If I want to use interrupts via the PLIC (i.e. using  MSS_UART_enable_irq() on UART0) instead the program (https://hastebin.com/aqodotenar.cpp) in not working, in the sense that the message which should be printed by MSS_UART_irq_tx() does not appear on the UART output, nor the input from the keyboard triggers the rx handler. Is there something else I am missing to be able to make it work? 

Thanks in advance!!!

**Comment by hughbreslin** [comment]
Hey @fcuzzocrea have you also enabled the PLIC interrupts and set the interrupt priority for the UART you're using? 

Have a look at this [example](https://github.com/polarfire-soc/polarfire-soc-bare-metal-library/blob/master/examples/mss-usb/mpfs-usb-host-hid/src/application/hart0/e51.c#L159) 

You can find IRQ names [here](https://github.com/polarfire-soc/polarfire-soc-bare-metal-library/blob/master/src/platform/mpfs_hal/common/mss_plic.h#L412) in the mss_plic.h file.

**Comment by hughbreslin** [comment, marked as answer]
Actually, sorry, [this](https://github.com/polarfire-soc/polarfire-soc-bare-metal-library/blob/master/examples/mss-gpio/mpfs-gpio-interrupt/src/application/hart0/e51.c#L120) example is much better!

**Comment by fcuzzocrea** [comment]
Thanks! I was missing to call PLIC_SetPriority() !

**Comment by hughbreslin** [comment]
If everything's working ok would you be able to mark the issue as closed? If you have any further issues on this please let me know :)

**Comment by fcuzzocrea** [comment]
Yeah of course! Thanks for you help!
